cfme/infrastructure/deployment_roles.py

```diff
@@ -1,67 +1,237 @@
+    toolbar = View.nested(DeploymentRoleToolbar)
+    entities = View.nested(DeploymentRoleEntities)
+    paginator = PaginationPane()
+
+    @property
+    def is_displayed(self):
+        """This is page currently being displayed"""
+        return (
+            self.in_dep_role and
+            (self.entities.title.text == 'Deployment Roles' or
+             self.entities.title.text == 'All Cluster / Deployment Role'))
+
+
+class DeploymentRoleAllForProviderView(DeploymentRoleView):
+    """The Deployment Role for Prover page"""
+    breadcrumb = BreadCrumb()
+    toolbar = View.nested(DeploymentRoleToolbar)
+    sidebar = View.nested(DeploymentRoleDetailsAccordion)
+    entities = View.nested(DeploymentRoleEntities)
+    paginator = PaginationPane()
+
+    @property
+    def is_displayed(self):
+        """Is this page currently being displayed"""
+        expected_title = '{} (All Deployment Roles)'.format(self.context['object'].provider.name)
+
+        return (
+            self.logged_in_as_current_user and
+            self.entities.breadcrumb.active_location == expected_title)
+
+
+class DeploymentRoleDetailsView(DeploymentRoleView):
+    """The details page for a Deployment Roles"""
+    toolbar = View.nested(DeploymentRoleDetailsToolbar)
+    sidebar = View.nested(DeploymentRoleDetailsAccordion)
+    entities = View.nested(DeploymentRoleDetailsEntities)
+
+    @property
+    def is_displayed(self):
+        """Is this page currently being displayed"""
+        expected_title = '{} (Summary)'.format(self.context['object'].name)
+        return (
+            self.in_dep_role and
+            self.entities.title.text == expected_title and
+            self.entities.breadcrumb.active_location == expected_title)
+
+
+class DeploymentRoleEditTagsView(DeploymentRoleView):
+    """The edit tags of Deployment Role"""
+    breadcrumb = BreadCrumb()
+    title = Text('#explorer_title_text')
+    select_tag = BootstrapSelect('tag_cat')
+    select_value = BootstrapSelect('tag_add')
+    save_button = Button('Save')
+    reset_button = Button('Reset')
+    cancel = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        """Is this page currently being displayed"""
+        return (
+            self.in_dep_role and
+            self.breadcrumb.active_location == 'Tag Assignment')
+
+
+class DeploymentRoleManagePoliciesView(DeploymentRoleView):
+    """Deployment role Manage Policies view."""
+    breadcrumb = BreadCrumb()
+    policies = BootstrapTreeview("protectbox")
+    save_button = Button("Save")
+    reset_button = Button("Reset")
+    cancel_button = Button("Cancel")
+
+    @property
+    def is_displayed(self):
+        """Is this page currently displayed"""
+        return (
+            self.in_dep_role and
+            (self.breadcrumb.active_location == "'Cluster / Deployment Role' Policy Assignment" or
+             self.breadcrumb.active_location == "'Deployment Role' Policy Assignment")
```

Comment:
Yes @quarckster , I found that sometimes, we have to face "Cluster /Deployment Role".